### PR TITLE
🪟 🎨 Add `PillSelect` UI component

### PR DESCRIPTION
**Pull request by edmundito**
## What
Related to #18186 
Figma: https://www.figma.com/file/liLQhcbpVHiEDW18kiXQe3/01_03_CONNECTIONS?node-id=2060%3A47239

Adds ads a new select component that appears as a pill button. Color and style variations are currently not supported (still in design discussion) and will be added in a future PR.

Screen Shot 2022-10-26 at 11 15 02
Screen Shot 2022-10-26 at 11 14 51

## How
* Implements the component with storybook
* Adds example to `Popout` storybook on how to use both together.
* Adds extra small `xs` size to `Text` component.

### CODE REVIEW

**Comment by timroes** [comment]
Looking at the designs this is used, this seems to be meant to be used as a form input i.e. a select? In which case this should rather be emulating a select element than a button element. Could you maybe clarify a bit what's the planned usage of this and whether this is really an interaction item (in which case popup would make sense) and not a form input (in which case we should rather go with either react-select or if we're fine with a limited set of functionality the List box from headless UI.

**Comment by edmundito** [comment]
@timroes, the intent was for this to be used alongside the `Popout` component, which allows it a select to render with any root component as the user needs to click. Still, after reading your comment, I noticed that the `Popout` usage is getting replaced by #17664. I could take advantage of the changes, but it looks like that component is specifically built for drop-down menus.

Looks like we have a lot of components that more or less do the same thing. This is not too different from select except for what it looks like.

**Comment by timroes** [comment]
The problem having those different components though is that some of them are not behaving like a select. So from my understanding we want to change e.g. the "sync mode" selection to be using this PillButton with a popup assigned? That would though render out in the form as a `button` with just some other popover HTML that will have more buttons, that we'll actually link up in JS to set form values then.

Since this is de facto a `select`, we should though make sure to render it as a select in the browser. Not doing so would break with accessibility, as well as we'd need to make sure all the keyboard interaction we'd expect it to have (e.g. type to jump to the option beginning with this letters) will need to be rebuild. That's why I believe we want to make sure to use react-select or headless UI Listbox here, to generate the right HTML we want.

If this button is still used somewhere else (purely as a button and not a select box, which the current property `asDropdown` suggests it might), we should render out different HTML in those places where it should be a button vs it should be a select, and maybe just share styles, but not the actual react component between them.

File: airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@use "scss/colors";
+
+.button {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: 3px 10px;
+  gap: 5px;
```
Comment (tealjulia) [inline airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss:8]: gap and padding can use spacing variables here

File: airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  border-radius: 20px;
```
Comment (tealjulia) [inline airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss:9]: is this worth adding a border-radius variable?  do we anticipate reusing this shape?

File: airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  cursor: pointer;
+  background: colors.$grey-50;
+  border: 1px solid colors.$grey-50;
```
Comment (tealjulia) [inline airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss:12]: can use border-thin variable

**Review by teallarson — COMMENTED** [review]
A few spots to migrate to using variables for spacing/borders.

I couldn't find an example in Figma of an open popout using this button.  Do we have one?  I had a few questions RE: the list element text size and popout list border radius for this dropdown style, but I couldn't find the design to match.

File: airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss (hunk 2 above)
Comment (edmundito) [reply airbyte-webapp/src/components/ui/PillButton/PillButton.module.scss]: Still need to finalize the border radius variables. I'll follow up with Nico and apply this later on.

**Comment by edmundito** [comment]
I haven't updated the title/description, but I pushed another change to turn this into a select component instead of a combination of components.

Popout is a convenient implementation on top of react-select, and I still found it useful for what's needed.

File: airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    },
+    buttonProps.className
+  );
+
+  return (
+    <button {...buttonProps} className={buttonClassName}>
+      <Text as="span" size="xs" className={styles.text}>
+        {children}
+      </Text>
+      <FontAwesomeIcon icon={faCaretDown} />
```
Comment (dizel852) [inline airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx:30]: Does `faCaretDown` icon should be visible always?  🤔

**Review by dizel852 — APPROVED** [review]
Code LGTM, tested locally.
PillButton + react select = 🔥 
I also tested keyboard interaction - seems like it works as should.

I guess [this](https://github.com/airbytehq/airbyte/pull/18376#discussion_r1004547404) comment was missed

File: airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx (hunk 4 above)
Comment (edmundito) [reply airbyte-webapp/src/components/ui/PillSelect/PillButton.tsx:30]: Yes, the PillButton now is just in support of the whole component so the caret always appears. I updated the description.

**Pull request merged**